### PR TITLE
Remove one level of hierarchy in STEP export

### LINKED ISSUE #1266: Eliminate level of hierarchy Assembly export
Assembly STEP export currently creates a root node with hardcoded name "CQ assembly". 

image

(from #1239 code with unsupported constraint removed)

This root node could be considered extraneous and be removed.
image

Regarding the root name,  I think the existing naming can be used here - the user specifies a name else it is auto generated uuid.

### PR DESCRIPTION
**Pull request by lorenzncode**
Do not hard code root name if unspecified

Resolves #1266

### CODE REVIEW

File: cadquery/occ_impl/assembly.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def toCAF(
     mesh: bool = False,
     tolerance: float = 1e-3,
     angularTolerance: float = 0.1,
-) -> Tuple[TDF_Label, TDocStd_Document]:
+) -> TDocStd_Document:
```
Comment (adam-urbanczyk) [inline cadquery/occ_impl/assembly.py:143]: Why change the signature, you can still return the top level label

Comment (lorenzncode) [reply cadquery/occ_impl/assembly.py:143]: I see.  I may not have  considered potential external usage of the function.   Internal usage was ignoring the label.  Yes, this change could be reverted to restore the original signature.

Comment (lorenzncode) [reply cadquery/occ_impl/assembly.py:143]: oops - not to mention breaking cq-editor.

Comment (lorenzncode) [reply cadquery/occ_impl/assembly.py]: I'm getting the label from XCAFPrs_DocumentExplorer, however, it could instead by set inside _toCAF if preferred.

Comment (adam-urbanczyk) [reply cadquery/occ_impl/assembly.py]: I'd propose to return it from `_toCAF` and do not use any additional methods.

**Review by jmwright — APPROVED** [review]
Thanks @lorenzncode !

**Pull request merged**